packages/@react-spectrum/s2/src/CoachMark.tsx

```diff
@@ -0,0 +1,530 @@
+  // If an action menu is present, place it next to the title.
+  gridTemplateColumns: {
+    default: ['1fr'],
+    ':has([data-slot=menu])': ['minmax(0, 1fr)', 'auto']
+  },
+  gridTemplateAreas: {
+    default: [
+      'title keyboard',
+      'description keyboard'
+    ],
+    ':has([data-slot=menu])': [
+      'title menu',
+      'keyboard keyboard',
+      'description description'
+    ]
+  },
+  columnGap: 4,
+  flexGrow: 1,
+  alignItems: 'baseline',
+  alignContent: 'space-between',
+  rowGap: {
+    size: {
+      XS: 4,
+      S: 4,
+      M: space(6),
+      L: space(6),
+      XL: 8
+    }
+  },
+  paddingTop: {
+    default: '--card-spacing',
+    ':first-child': 0
+  },
+  paddingBottom: {
+    default: '[calc(var(--card-spacing) * 1.5 / 2)]',
+    ':last-child': 0
+  }
+});
+
+let actionMenu = style({
+  gridArea: 'menu',
+  // Don't cause the row to expand, preserve gap between title and description text.
+  // Would use -100% here but it doesn't work in Firefox.
+  marginY: '[calc(-1 * self(height))]'
+});
+
+let footer = style({
+  display: 'flex',
+  flexDirection: 'row',
+  alignItems: 'end',
+  justifyContent: 'space-between',
+  gap: 8,
+  paddingTop: '[calc(var(--card-spacing) * 1.5 / 2)]'
+});
+
+const actionButtonSize = {
+  XS: 'XS',
+  S: 'XS',
+  M: 'S',
+  L: 'M',
+  XL: 'L'
+} as const;
+
+export const CoachMarkContext = createContext<ContextValue<CoachMarkProps, HTMLElement>>({});
+
+export const CoachMark = forwardRef((props: CoachMarkProps, ref: ForwardedRef<HTMLElement>) => {
+  let colorScheme = useContext(ColorSchemeContext);
+  [props, ref] = useContextProps(props, ref, CoachMarkContext);
+  let {UNSAFE_style} = props;
+  let {size = 'M'} = props;
+  let popoverRef = useObjectRef(ref);
+
+  let children = (
+    <Provider
+      values={[
+        [ImageContext, {alt: '', styles: image}],
+        [TextContext, {
+          slots: {
+            [DEFAULT_SLOT]: {},
+            title: {styles: title({size})},
+            description: {styles: description({size})},
+            steps: {styles: steps}
+          }
+        }],
+        [KeyboardContext, {styles: keyboard}],
+        [ContentContext, {styles: content({size})}],
+        [DividerContext, {size: 'S'}],
+        [FooterContext, {styles: footer}],
+        [ActionMenuContext, {
+          isQuiet: true,
+          size: actionButtonSize[size],
+          // @ts-ignore
+          'data-slot': 'menu',
+          styles: actionMenu
+        }]
+      ]}>
+      <ImageCoordinator>
+        {props.children}
+      </ImageCoordinator>
+    </Provider>
+  );
+
+  return (
+    <AriaPopover
+      {...props}
+      ref={popoverRef}
+      style={{
+        ...UNSAFE_style,
+        // Override default z-index from useOverlayPosition. We use isolation: isolate instead.
+        zIndex: undefined
+      }}
+      className={(renderProps) =>  mergeStyles(popover({...renderProps, colorScheme}))}>
+      <Card>
```

Comment:
I guess you're planning to change this later?

Reply:
yeah, in my follow up i just have CoachMark as an empty container
https://github.com/adobe/react-spectrum/pull/8095/files#diff-bf811e14d299f5b052ff23dae48c30e7808ab7d09672bc60eeac784952161d58R153